components/camel-huaweicloud-smn/src/main/java/org/apache/camel/component/huaweicloud/smn/SimpleNotificationProducer.java

```diff
@@ -0,0 +1,366 @@
+
+import java.util.HashMap;
+
+import com.huaweicloud.sdk.core.auth.BasicCredentials;
+import com.huaweicloud.sdk.core.http.HttpConfig;
+import com.huaweicloud.sdk.smn.v2.SmnClient;
+import com.huaweicloud.sdk.smn.v2.model.PublishMessageRequest;
+import com.huaweicloud.sdk.smn.v2.model.PublishMessageRequestBody;
+import com.huaweicloud.sdk.smn.v2.model.PublishMessageResponse;
+import org.apache.camel.Exchange;
+import org.apache.camel.component.huaweicloud.smn.constants.SmnConstants;
+import org.apache.camel.component.huaweicloud.smn.constants.SmnOperations;
+import org.apache.camel.component.huaweicloud.smn.constants.SmnServices;
+import org.apache.camel.component.huaweicloud.smn.models.ClientConfigurations;
+import org.apache.camel.support.DefaultProducer;
+import org.apache.camel.util.ObjectHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SimpleNotificationProducer extends DefaultProducer {
+    private static final Logger LOG = LoggerFactory.getLogger(SimpleNotificationProducer.class);
+    private SimpleNotificationEndpoint simpleNotificationEndpoint;
+    private SmnClient smnClient;
+    private ClientConfigurations clientConfigurations;
+
+    public SimpleNotificationProducer(SimpleNotificationEndpoint endpoint) {
+        super(endpoint);
+        this.simpleNotificationEndpoint = endpoint;
+    }
+
+    public void process(Exchange exchange) throws Exception {
+
+        /**
+         * the produce method is invoked when a message arrives at producer endpoint (lazy init of SmnClient on first
+         * message)
+         */
+        if (this.smnClient == null) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Initializing the SmnClient");
+            }
+            validateAndInitializeSmnClient(simpleNotificationEndpoint);
+        }
+
+        String service = simpleNotificationEndpoint.getSmnService();
+
+        if (!ObjectHelper.isEmpty(service)) {
+            switch (service) {
+                case SmnServices.PUBLISH_MESSAGE:
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("Using message publishing service");
+                    }
+                    performPublishMessageServiceOperations(simpleNotificationEndpoint, exchange);
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("Completed publishing message");
+                    }
+                    break;
+                default:
+                    if (LOG.isErrorEnabled()) {
+                        LOG.error("Unsupported service name {}", service);
+                    }
+                    throw new UnsupportedOperationException(String.format("service %s is not a supported service", service));
+            }
+        } else {
+            if (LOG.isErrorEnabled()) {
+                LOG.error("Service name is null/empty");
+            }
+            throw new IllegalStateException("service name cannot be null/empty");
```

Comment:
out of curiosity, do you think it just makes sense to have a default service instead of throwing an error? especially that there is currently one service implemented

Reply:
It was just a personal choice. I have removed the lazy initialization to class init time